packages/pages/src/vite-plugin/build/closeBundle/functionMetadata.ts

```diff
@@ -0,0 +1,63 @@
+import fs from "fs";
+import path from "path";
+import esbuild from "esbuild";
+import { importFromString } from "module-from-string";
+import glob from "glob";
+
+const FUNCTIONS_PATH = path.resolve("functions");
+const FUNCTION_METADATA_PATH = path.join(
+  FUNCTIONS_PATH,
+  "functionMetadata.json"
+);
+const TEMP_DIR = ".temp";
+
+/** Metadata for a serverless function. */
+type FunctionMetadata = {
+  /** Name of the function. */
+  entrypoint: string;
+};
+
+/**
+ * Returns a mapping of file path (relative to the repo root) to the metadata
+ * for the function that is the default export of that file.
+ */
+const getFunctionMetadataMap = async (): Promise<
+  Record<string, FunctionMetadata>
+> => {
+  const filepaths = glob
+    .sync(path.join(FUNCTIONS_PATH, "**/*.{tsx,jsx,js,ts}"), { nodir: true })
```

Comment:
For my edification, do we allow TSX or JSX server-less functions? What's the use-case there?

Reply:
I originally included TSX and JSX because I wasn't sure of all the envisioned use cases for serverless functions, but Andrew confirmed that only JS or TS would be used. Updated!